### PR TITLE
feature(Pipeline Expressions): Add function pipelineIdInApplication hich allows to fetch the ID of a pipeline given its name and its application name

### PR DESCRIPTION
**Pull request by AbdulRahmanAlHamali**
This function is useful when we are running a dependent pipeline that falls in a different application

### CODE REVIEW

File: orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/pipeline/PipelineExpressionFunctionProvider.java [hunk 1]
```diff
@@ -83,39 +92,18 @@ public static String pipelineId(PipelineExecution execution, String pipelineName
           "pipelineName must be specified for function #pipelineId");
     }
```
Comment (AbdulRahmanAlHamali) [inline orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/pipeline/PipelineExpressionFunctionProvider.java:94]: I extracted the logic here into a common function

**Comment by AbdulRahmanAlHamali** [comment]
It's sad that we cannot overload pipeline expression functions

**Comment by zachsmith1** [comment]
@AbdulRahmanAlHamali Please merge master into your branch in order to complete this PR.

**Comment by AbdulRahmanAlHamali** [comment]
Done :)

**Pull request merged**